src/index.js

```diff
@@ -225,32 +245,50 @@ const runMergeScript = async (
       const tempDatabase = `${flags.db}_${key}`;
       const targetDatabase = `${flags.newdb || flags.db}`;
 
-      console.info(`  INFO: Merging ${tempDatabase} to ${targetDatabase}...`);
-
-      const run = async () => {
-        await executeCommand(
-          `SELECT * INTO ${targetDatabase}..:MEASUREMENT FROM ${tempDatabase}../.*/ GROUP BY *;`,
+      const measurements =
+        (flags.measurements && flags.measurements.split(';')) || [];
+      const fields = (flags.fields && flags.fields.split(';')) || [];
+      if (measurements.length !== fields.length && fields.length > 0) {
+        throw new Error(
+          'measurement and field list need to be equal size or fields should be empty',
         );
-        await executeCommand(`DROP DATABASE ${tempDatabase}`);
-
-        console.info(`  INFO: Merged ${tempDatabase} to ${targetDatabase}!`);
-        return Promise.resolve();
-      };
-
-      return limit(() =>
-        pRetry(run, {
-          onFailedAttempt: error => {
-            console.log(
-              `Attempt ${
-                error.attemptNumber
-              } for ${tempDatabase} to ${targetDatabase} failed. There are ${
-                error.attemptsLeft
-              } attempts left.`,
+      }
+      if (!measurements.length) {
+        measurements.push(null);
+        fields.push(null);
+      }
+
+      return Promise.all(
+        measurements.map((measurement, i) => {
+          const run = async () => {
+            await executeCommand(
+              `SELECT ${fields[i]} INTO ${targetDatabase}..${measurement ||
+                `:MEASUREMENT`} FROM  ${tempDatabase}..${measurement ||
+                `/.*/`} GROUP BY *`,
+            );
+
+            console.info(
+              `  INFO: Merged ${measurement ||
+                `*`} ON ${tempDatabase} to ${targetDatabase}!`,
             );
-          },
-          retries: 5,
+            return Promise.resolve();
```

Comment:
Nit: async functions are wrapped in a promise so this is not needed as nothing is returned